tests/baselines/reference/inferTypes2.js

```diff
@@ -52,3 +52,29 @@ export declare function foo2<T>(obj: T): T extends {
 export declare function bar2<T>(obj: T): T extends {
     x: infer P extends number ? infer P : string;
 } ? P : never;
+
+
+//// [DtsFileErrors]

```

Comment:
There was a discussion about this thing when this test case was introduced, see [here](https://github.com/microsoft/TypeScript/pull/22764#pullrequestreview-105846102). Does it meant that this PR is obsolete and the current behavior is OK? It certainly feels like a bug to the reporter and it felt like one to me when I've read the issue.

It turns out though that @ahejlsberg explicitly reverted a similar strict containment rule in that old PR [here](https://github.com/microsoft/TypeScript/pull/22764/commits/a75449cabc70de2ee74d7d9e4d37847d0e357f1f). Would the issue resolution be different in 2024 than it was in 2018? 😅 

cc @weswigham